### PR TITLE
LM-13: add project structure

### PR DESCRIPTION
**Pull request by rgillen-uno**
PR for merging all approved changes for Project structure PRs into develop.

### CODE REVIEW

File: AdminCore.Services.Tests/ProjectServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+      GetMockedResourcesProjectRepo(new List<Project>(), out var mapper, out var projectService, out var ormContext);
+
+      // Act
+      var serviceActual = projectService.GetProjects();
+
+      // Assert
+      ormContext.Received(1).ProjectRepository.Get(Arg.Any<Expression<Func<Project, bool>>>(),
+        Arg.Any<Func<IQueryable<Project>, IOrderedQueryable<Project>>>(),
+        Arg.Any<Expression<Func<Project, object>>[]>());
+//      mapper.Received(1).Map<IList<ProjectDto>>(Arg.Is<IList<Project>>(x => x != null && x.Count == 0));
```
Comment (kkuczkowski-unosq) [inline AdminCore.Services.Tests/ProjectServiceTests.cs]: what's up with all these commented assertions?

File: AdminCore.WebApp/Properties/launchSettings.json [hunk 2]
```diff
@@ -18,11 +18,11 @@
     },
     "AdminCore.WebApp": {
       "commandName": "Project",
-      "launchBrowser": true,
+      "launchBrowser": false,
```
Comment (kkuczkowski-unosq) [inline AdminCore.WebApp/Properties/launchSettings.json:21]: why this change?

File: AdminCore.WebApp/Controllers/ProjectController.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  public class ProjectController : BaseController
+  {
+    private readonly IProjectService _projectService;
+
+    public ProjectController(IProjectService projectService, IMapper mapper) : base(mapper)
+    {
+      _projectService = projectService;
+    }
+
+    private IActionResult HandleBadRequest(Func<IActionResult> serviceAction)
```
Comment (kkuczkowski-unosq) [inline AdminCore.WebApp/Controllers/ProjectController.cs]: private function should probably not live between public functions, more suitable place would be bottom of the class.

File: AdminCore.DatabaseMigration/db/migrations/V1.12__add_project.sql [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    deleted boolean NOT NULL,
+
+    CONSTRAINT project_pkey PRIMARY KEY (project_id),
+
+    CONSTRAINT project_project_parent_id_fkey FOREIGN KEY (project_parent_id)
+      REFERENCES public.project (project_id) MATCH SIMPLE
+      ON UPDATE NO ACTION
+      ON DELETE NO ACTION,
+    CONSTRAINT project_client_id_fkey FOREIGN KEY (client_id)
+        REFERENCES public.client (client_id) MATCH SIMPLE
```
Comment (kkuczkowski-unosq) [inline AdminCore.DatabaseMigration/db/migrations/V1.12__add_project.sql]: indent here is a bit off

**Review by kkuczkowski-unosq — COMMENTED** [review]
looks good

File: AdminCore.WebApp/Properties/launchSettings.json (hunk 2 above)
Comment (rgillen-uno) [reply AdminCore.WebApp/Properties/launchSettings.json:21]: This launches your system default browser at the defined host path when you run the app in an ide i.e localhost:8081

I didn't think this was necessary

Comment (kkuczkowski-unosq) [reply AdminCore.WebApp/Properties/launchSettings.json:21]: I know what it does, this just seemed out of context. That's fair enough.

**Review by kkuczkowski-unosq — APPROVED** [review]
ship it

**Comment by kkuczkowski-unosq** [comment]
make sure you merge develop into this branch before merging, I just merged LM-40
